Data/mne

ScrabbleDestroyer opened the pull request:
Opening pull request (to develop), because I want to see what happens :).

The data loading module now returns list of custom class objects, instead of tuples with things and things. If returned list is returned_list, then the EEG accessible from returned_list[0][0] should now be at returned_list[0].raw_eeg, and patient evaluation at returned_list[0].evaluation, instead of returned_list[0][1]. You code will have to be adjusted though. The data can be filtered at different points and / or have power spectrum calculated via mne library.

pfigiel commented on cvapr_data.py:
```diff
@@ -67,20 +83,89 @@ def _load_evals(eval_path):
+        if high_freq is None:
+            high_freq = self._high_freq
+        sample_freq = self.mne_eeg.info["sfreq"]
+        return psd.psd_welch(self.mne_eeg, self._low_freq, self._high_freq, n_fft = round(sample_freq / step))
```
I believe it should be (self.mne_eeg, **low_freq, high_freq**...). Otherwise we can't control these parameters through function arguments. Also with this line as is, an exception is thrown when using this function.

ScrabbleDestroyer replied:
Fixed this and some other things i've missed previously due to this unnoticed error.

ScrabbleDestroyer committed:
Fixed wrong parameters passed to power spectrum

MNE power spectrum functions always require frequencies range, just they are 0 and inf by default, not None. Fixed it and adapted interface. Also fixed unused parameters in PictureBlockData power_spectrum.

ScrabbleDestroyer committed:
Merge branch 'develop' into data/mne
